lib/Runtime/Language/AsmJsByteCodeGenerator.cpp

```diff
@@ -1934,45 +1921,43 @@ namespace Js
         return emitInfo;
     }
 
-    EmitExpressionInfo AsmJSByteCodeGenerator::EmitMathBuiltin(ParseNode* pnode, AsmJsMathFunction* mathFunction, AsmJsRetType expectedType)
+    EmitExpressionInfo AsmJSByteCodeGenerator::EmitMathBuiltin(ParseNode* pnode, AsmJsMathFunction* mathFunction)
     {
         if (mathFunction->GetMathBuiltInFunction() == AsmJSMathBuiltinFunction::AsmJSMathBuiltin_max || mathFunction->GetMathBuiltInFunction() == AsmJSMathBuiltinFunction::AsmJSMathBuiltin_min)
         {
-            return EmitMinMax(pnode, mathFunction, expectedType);
+            return EmitMinMax(pnode, mathFunction);
         }
 
         ++mNestedCallCount;
 
         const ArgSlot argCount = pnode->sxCall.argCount;
         ParseNode* argNode = pnode->sxCall.pnodeArgs;
+        const bool isFRound = AsmJsMathFunction::IsFround(mathFunction);
 
         // for fround, if we have a fround(NumericLiteral), we want to just emit Ld_Flt NumericLiteral
-        if (argCount == 1 && IsFRound(mathFunction) && ParserWrapper::IsFroundNumericLiteral(argNode))
+        if (argCount == 1 && isFRound && ParserWrapper::IsFroundNumericLiteral(argNode))
         {
-            Assert(expectedType == AsmJsRetType::Float);
             StartStatement(pnode);
             RegSlot dst = mFunction->AcquireTmpRegister<float>();
-            EmitExpressionInfo emitInfo(dst, expectedType.toType());
+            EmitExpressionInfo emitInfo(dst, AsmJsType::Float);
```

Comment:
This is the force to Float type in Fround case making the previous bad check useless